### PR TITLE
Patch/copyrightyear update

### PR DESCRIPTION
**Pull request by SGSSGene**
Update all license text to SPDX-headers  and include a new CI job, which checks for valid SPDX information.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# SPDX-FileCopyrightText: 2006-2024 Knut Reinert & Freie Universität Berlin
+# SPDX-FileCopyrightText: 2016-2024 Knut Reinert & MPI für molekulare Genetik
+# SPDX-License-Identifier: CC0-1.0
```
Comment (eseiler) [inline .gitmodules:3]: Does this need a new line? The submodules aren't checked out in the ci

Comment (SGSSGene) [reply .gitmodules]: Yeah, this doesn't seem to work like this at all :/ added a .gitmodules.license

Comment (eseiler) [reply .gitmodules:3]: Looks like some commit removed the `submodules/*` directories. I tried readding them

**Review by eseiler — APPROVED** [review]

**Comment by eseiler** [comment]
We should also update our `LICENSE.md` and check the install/package script. But this should be a follow-up :)

**Pull request merged**